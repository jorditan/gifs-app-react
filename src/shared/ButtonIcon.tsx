import { type LucideProps } from "lucide-react"

interface Props {
  tooltipText: string
  icon: React.ForwardRefExoticComponent<Omit<LucideProps, "ref"> & React.RefAttributes<SVGSVGElement>>
  variant: "outline" | "link" | "default" | "destructive" | "secondary" | "ghost" | null | undefined;
}

import {
  Tooltip,
  TooltipContent,
  TooltipTrigger,
} from "@/components/ui/tooltip"

import { Button } from "@/components/ui/button"

export function ButtonIcon(props: Props) {
  return (
    <Tooltip >
      <TooltipTrigger>
        <Button variant={props.variant} size="icon" className="size-8">
          <props.icon />
        </Button>
      </TooltipTrigger>
      <TooltipContent>
        {props.tooltipText}
      </TooltipContent>
    </Tooltip>
  )
}
