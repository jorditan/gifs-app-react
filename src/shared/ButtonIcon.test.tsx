import { fireEvent, render, screen } from "@testing-library/react"
import { describe, expect, test, vi } from "vitest"
import { ButtonIcon } from "./ButtonIcon"
import { Download } from "lucide-react";


describe('ButtonIcon', () => {
  test('should render ButtonIcon correctly', () => {
    const { container } = render(<ButtonIcon tooltipText={""} icon={Download} handleAction={() => { }} variant={"default"} />)

    expect(container).toMatchSnapshot()
    expect(screen.getByRole('button')).toBeDefined();
  })

  test('should handleAction callend when button is clicked', () => {
    const handleAction = vi.fn();

    render(<ButtonIcon tooltipText={""} icon={Download} handleAction={handleAction} variant={"default"} />)

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(handleAction).toBeCalled();
  })

  // test('should render tooltip content correctly', async () => {
  //   const tooltipContent = "Hola soy un tooltip"
  //   render(<ButtonIcon tooltipText={tooltipContent} icon={Download} handleAction={() => { }} variant={"default"} />)
  //   expect(await screen.findByRole('tooltip')).toHaveTextContent(tooltipContent);
  // })
})