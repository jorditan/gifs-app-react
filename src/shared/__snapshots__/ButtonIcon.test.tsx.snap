// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonIcon > should render ButtonIcon correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground hover:bg-primary/90 size-9"
    data-slot="tooltip-trigger"
    data-state="closed"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-download"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 15V3"
      />
      <path
        d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
      />
      <path
        d="m7 10 5 5 5-5"
      />
    </svg>
  </button>
</div>
`;
